modular_skyrat/master_files/code/modules/pain_shock/status_indicator.dm

```diff
@@ -0,0 +1,198 @@
+			"confused",
+			"blinded",
+			"critical")
+		for(potential_indicator in overlays)
+			cut_overlay(potential_indicator)
+
+	else
+		for(var/thing in status_indicators)
+			cut_overlay(thing)
+
+// Refreshes the indicators over a mob's head. Should only be called when adding or removing a status indicator with the above procs,
+// or when the mob changes size visually for some reason.
+/mob/living/proc/handle_status_indicators()
+	// First, get rid of all the overlays.
+	cut_indicators_overlays()
+
+	if(!LAZYLEN(status_indicators))
+		return
+
+	if(stat == DEAD)
+		return
+	var/mob/living/carbon/carbon = src
+	var/icon_scale = get_icon_scale(carbon)
+	// Now put them back on in the right spot.
+	var/our_sprite_x = 16 * icon_scale
+	var/our_sprite_y = 24 * icon_scale
+
+	var/x_offset = our_sprite_x // Add your own offset here later if you want.
+	var/y_offset = our_sprite_y + STATUS_INDICATOR_Y_OFFSET
+
+	// Calculates how 'long' the row of indicators and the margin between them should be.
+	// The goal is to have the center of that row be horizontally aligned with the sprite's center.
+	var/expected_status_indicator_length = (STATUS_INDICATOR_ICON_X_SIZE * status_indicators.len) + (STATUS_INDICATOR_ICON_MARGIN * max(status_indicators.len - 1, 0))
+	var/current_x_position = (x_offset / 2) - (expected_status_indicator_length / 2)
+
+	// In /mob/living's `update_transform()`, the sprite is horizontally shifted when scaled up, so that the center of the sprite doesn't move to the right.
+	// Because of that, this adjustment needs to happen with the future indicator row as well, or it will look bad.
+	current_x_position -= (32 / 2) * (icon_scale - 1)
+
+	// Now the indicator row can actually be built.
+	for(var/thing in status_indicators)
+		var/image/I = thing
+
+		// This is a semi-HUD element, in a similar manner as medHUDs, in that they're 'above' everything else in the world,
+		// but don't pierce obfuscation layers such as blindness or darkness, unlike actual HUD elements like inventory slots.
+		I.plane = PLANE_STATUS
+		I.layer = STATUS_LAYER
+		I.appearance_flags = PIXEL_SCALE|TILE_BOUND|NO_CLIENT_COLOR|RESET_COLOR|RESET_ALPHA|RESET_TRANSFORM|KEEP_APART
+		I.pixel_y = y_offset
+		I.pixel_x = current_x_position
+		add_overlay(I)
+		// Adding the margin space every time saves a conditional check on the last iteration,
+		// and it won't cause any issues since no more icons will be added, and the var is not used for anything else.
+		current_x_position += STATUS_INDICATOR_ICON_X_SIZE + STATUS_INDICATOR_ICON_MARGIN
+
+/mob/living/proc/get_icon_scale(carbon)
+	if(!iscarbon(carbon)) // normal mobs are always 1 for scale - not borg compatible but ok
+		return 1
+	var/mob/living/carbon/passed_mob = carbon // we're possibly a player! We have size prefs!
+	var/mysize = (passed_mob.dna.current_body_size ? passed_mob.dna.current_body_size : 1)
+	return mysize
+
+/atom/movable/screen/plane_master/runechat/status
+	name = "status plane master"
```

Comment:
Is there any reason why this is a subtype of `/plane_master/runechat`?

Also, rename it to be clearer:
```suggestion
/atom/movable/screen/plane_master/runechat/status
	name = "status indicator plane master"
```